### PR TITLE
gh-91266: refactor bytearray strip methods

### PR DESCRIPTION
**Pull request by eendebakpt**
The bytearray strip, lstrip and rstrip methods contain duplicated code. This PR refactors the three implementations to use common code.

https://bugs.python.org/issue47110

### CODE REVIEW

**Review by MaxwellDupre — APPROVED** [review]
Test Results:
415 tests OK.

8 tests failed:
    test_asyncio test_bdb test_distutils test_embed test_idle
    test_ossaudiodev test_peg_generator test_venv

Looks like none relevant.

**Comment by sweeneyde** [comment]
The Python-visible API probably shouldn't change.

**Comment by eendebakpt** [comment]
@sweeneyde Can you review again? Thanks

**Review by sweeneyde — APPROVED** [review]
We should typically be pretty conservative about refactoring PRs, but since this eliminates 45 lines, I think it makes sense.

**Comment by peendebak** [comment]
@sweeneyde Thanks. Should I add a new entry for this?

**Comment by sweeneyde** [comment]
Sure, yes: better to err on the side of documenting everything.

File: Misc/NEWS.d/next/Core and Builtins/2022-04-13-07-14-30.gh-issue-91266.6Vkzzt.rst [hunk 1]
```diff
@@ -0,0 +1 @@
+Refactor the `bytearray` strips method `strip`, `lstrip` and `rstrip` to use a common implementation.
```
Comment (sweeneyde) [inline Misc/NEWS.d/next/Core and Builtins/2022-04-13-07-14-30.gh-issue-91266.6Vkzzt.rst]:
```suggestion
Refactor the ``bytearray`` strip methods ``strip``, ``lstrip`` and ``rstrip`` to use a common implementation.
```

**Pull request merged**